src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -229,13 +227,14 @@ public static class ResolverUtility
         /// <param name="remoteWalkContext">remote Providers (all sources, including file sources)</param>
         /// <param name="cancellationToken">cancellation token</param>
         /// <returns>The requested range and remote match.</returns>
-        public static Task<Tuple<LibraryRange, RemoteMatch>> FindPackageLibraryMatchCachedAsync(
-            ConcurrentDictionary<LibraryRange, Task<Tuple<LibraryRange, RemoteMatch>>> cache,
+        public static async Task<Tuple<LibraryRange, RemoteMatch>> FindPackageLibraryMatchCachedAsync(
```

Comment:
If possible, consider reverting the async part.

Reply:
Done